pkg/didcomm/protocol/issuecredential/states.go

```diff
@@ -288,8 +300,19 @@ func (s *offerReceived) CanTransitionTo(st state) bool {
 		st.Name() == stateNameAbandoning
 }
 
-func (s *offerReceived) ExecuteInbound(_ *metaData) (state, stateAction, error) {
-	return nil, nil, fmt.Errorf("%s: ExecuteInbound is not implemented yet", s.Name())
+func (s *offerReceived) ExecuteInbound(md *metaData) (state, stateAction, error) {
+	if !isDataCorrect(md.Msg) {
```

Comment:
I know you left a TODO for `isDataCorrect`, but not returning an error if `!isDataCorrect` is really confusing. Where are we going with this?

Reply:
According to the diagram, if data is not correct, it is fine. In that case, we are going back to `proposalSent`  state and we are sending a proposal message.

Reply:
It looks like a negotiation, we may receive an offer and then go back and send a proposal and we can do it continuously until data is correct or someone of the participants stops the protocol. My question here what actually is the correct data, how to check it? For now, I have added this `0000-0000` id which allows me to mock this logic for the tests.